libresapi/src/api/FileSharingHandler.cpp

```diff
@@ -40,7 +42,30 @@ FileSharingHandler::FileSharingHandler(StateTokenServer *sts, RsFiles *files):
 
 	addResourceHandler("download", this, &FileSharingHandler::handleDownload);
 
-	mStateToken = mStateTokenServer->getNewToken();
+	mLocalDirStateToken = mStateTokenServer->getNewToken();
+	mRemoteDirStateToken = mStateTokenServer->getNewToken();
+	mNotify->registerNotifyClient(this);
+}
+
+FileSharingHandler::~FileSharingHandler()
+{
+	mNotify->unregisterNotifyClient(this);
+}
+
+void FileSharingHandler::notifyListChange(int list, int /* type */)
+{
+	if(list == NOTIFY_LIST_DIRLIST_LOCAL)
+	{
+		RsStackMutex stack(mMtx); /********** STACK LOCKED MTX ******/
```

Comment:
You should use RS_STACK_MUTEX(mMtx) macro instead, also the comment is unnecessary